File: 199-binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        que = deque()
        if root : que.append(root)

        while len(que)>0:
            plevel = [] # level list left and right always 
            for i in range(len(que)):
                ele = que.popleft()
                if ele.left: # here we can avoid this 
                    que.append(ele.left)
                if ele.right:
                    que.append(ele.right)
                plevel.append(ele.val)
            ans.append(plevel[-1])
        return ans
            
        

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        if (!root){return vector<int>{};}
        queue<TreeNode*>q;
        vector<int>ans;
        q.push(root);
        while(!q.empty()){
            vector<int>plist;
            int q_size = q.size();
            for(int i=0;i<q_size;++i){
                TreeNode *curr = q.front();
                q.pop();
                plist.push_back(curr->val);
                if (curr->left){q.push(curr->left);}
                if (curr->right){q.push(curr->right);}
            }
            //pushing last element is key usally we store right elements in plist rest all is bfs
            ans.push_back(plist[plist.size()-1]);
        }
    return ans;
    }
    
};